resolvers.js

```diff
@@ -526,6 +535,37 @@ const Resolvers = {
         logger.error(e.message);
         return e.message;
       }
+    },
+
+    async deleteOpticalImage(_, args) {
+      let {datasetId} = args;
+      const payload = jwt.decode(args.jwt, config.jwt.secret);
+      const basePath = `http://localhost:${config.img_storage_port}`;
+      try {
+        await checkPermissions(datasetId, payload);
+        let opticalImageIds = await pg.select('id').from('optical_image').where('ds_id', '=', datasetId);
+        let rawOpticalImage = await pg.select('optical_image').from('dataset').where('id','=',datasetId);
+        Array.prototype.push.apply(opticalImageIds, rawOpticalImage);
+        opticalImageIds.forEach((Image) => {
```

Comment:
I'd change the logic here a little bit to reduce the code duplication.

how it works now:

1. fetch IDs of optical images
2. fetch ID of raw image
3. merge the arrays of IDs into 1 array
4. iterate over the array, and depending on the type of image ID send `DELETE` request to a specific URL

how I'd put it:

1. fetch IDs of optical images
2. generate an array of URLs to send `DELETE` request to
3. fetch ID of raw image
4. generate 1 more URL to send `DELETE` request to
5. send `DELETE` requests to each of generated URLs

Reply:
Thanks @iprotsyuk , I will send changes with a new commit (coming)